src/mscorlib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2230,7 +2338,7 @@ private unsafe void WriteEventString(EventLevel level, long keywords, string msg
                         data.Ptr = (ulong)msgStringPtr;
                         data.Size = (uint)(2 * (msgString.Length + 1));
                         data.Reserved = 0;
-                        m_provider.WriteEvent(ref descr, null, null, 1, (IntPtr)((void*)&data));
+                        m_provider.WriteEvent(ref descr, IntPtr.Zero, null, null, 1, (IntPtr)((void*)&data));
```

Comment:
I see that we don't crash (we have a check), but it also means we don't get any diagnostic information (these events are dropped).   Thus my suggestion above still stands.